framework/src/abi/schema.ts

```diff
@@ -0,0 +1,923 @@
+								},
+							},
+						},
+					},
+				},
+				logger: {
+					fieldNumber: 3,
+					type: 'object',
+					properties: {
+						consoleLogLevel: {
+							fieldNumber: 1,
+							dataType: 'string',
+						},
+						fileLogLevel: {
+							fieldNumber: 2,
+							dataType: 'string',
+						},
+					},
+				},
+				genesis: {
+					fieldNumber: 4,
+					type: 'object',
+					properties: {
+						communityIdentifier: {
+							fieldNumber: 1,
+							dataType: 'string',
+						},
+						maxTransactionsSize: {
+							fieldNumber: 2,
+							dataType: 'uint32',
+						},
+						maxFeePerByte: {
+							fieldNumber: 3,
+							dataType: 'uint32',
+						},
+						blockTime: {
+							fieldNumber: 4,
+							dataType: 'uint32',
+						},
+						bftBatchSize: {
+							fieldNumber: 5,
+							dataType: 'uint32',
+						},
+					},
+				},
+				network: {
+					fieldNumber: 5,
+					type: 'object',
+					properties: {
+						port: {
+							fieldNumber: 1,
+							dataType: 'uint32',
+						},
+						hostIP: {
+							fieldNumber: 2,
+							dataType: 'string',
+						},
+						seedPeers: {
+							fieldNumber: 3,
+							...networkPeerSchema,
+						},
+						fixedPeers: {
+							fieldNumber: 4,
+							...networkPeerSchema,
+						},
+						whitelistedPeers: {
+							fieldNumber: 5,
+							...networkPeerSchema,
+						},
+						blackListedIPs: {
+							fieldNumber: 6,
+							type: 'array',
+							items: {
+								dataType: 'string',
+							},
+						},
+						maxOutboundConnections: {
+							fieldNumber: 7,
+							dataType: 'uint32',
+						},
+						maxInboundConnections: {
+							fieldNumber: 8,
+							dataType: 'uint32',
+						},
+						advertiseAddress: {
+							fieldNumber: 9,
+							dataType: 'boolean',
+						},
+					},
+				},
+				txpool: {
+					fieldNumber: 6,
+					type: 'object',
+					properties: {
+						maxTransactions: {
+							fieldNumber: 1,
+							dataType: 'uint32',
+						},
+						maxTransactionsPerAccount: {
+							fieldNumber: 2,
+							dataType: 'uint32',
+						},
+						transactionExpiryTime: {
+							fieldNumber: 3,
+							dataType: 'uint32',
+						},
+						minEntranceFeePriority: {
+							fieldNumber: 4,
+							dataType: 'uint32',
+						},
+						minReplacementFeeDifference: {
+							fieldNumber: 5,
+							dataType: 'uint32',
+						},
+					},
+				},
+				generator: {
+					fieldNumber: 7,
+					type: 'object',
+					properties: {
+						password: {
+							fieldNumber: 1,
+							dataType: 'string',
+						},
+						force: {
+							fieldNumber: 2,
+							dataType: 'boolean',
+						},
+						keys: {
+							fieldNumber: 3,
+							type: 'array',
+							items: {
+								type: 'object',
+								properties: {
+									address: {
+										fieldNumber: 1,
+										dataType: 'bytes',
+									},
+									encryptedPassphrase: {
+										fieldNumber: 2,
+										dataType: 'string',
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+};
```

Comment:
can we some how make separate schema for txPool, logger etc and refer here so that we can avoid this huge schema